packages/server/src/workers/expand.ts

```diff
@@ -0,0 +1,156 @@
+import { CodeSystem, ValueSet } from '@medplum/fhirtypes';
+import { Job, Queue, QueueBaseOptions, Worker } from 'bullmq';
+import { MedplumServerConfig } from '../config';
+import { getRequestContext, tryRunInRequestContext } from '../context';
+import { globalLogger } from '../logger';
+import { validateCodings } from '../fhir/operations/codesystemvalidatecode';
+import { OperationOutcomeError, badRequest, mapFilter } from '@medplum/core';
+import { Column, InsertQuery, Literal, SelectQuery } from '../fhir/sql';
+import { DatabaseMode, getDatabasePool } from '../database';
+import { expansionQuery } from '../fhir/operations/expand';
+
+/*
+ * The expand worker constructs and stores the full expansion of ValueSet resources,
+ * to improve performance of typeahead $expand queries.
+ */
+
+export interface ExpandJobData {
+  readonly valueSet: ValueSet;
+  readonly codeSystems: Record<string, CodeSystem>;
+  readonly requestId?: string;
+  readonly traceId?: string;
+}
+
+const queueName = 'ExpandQueue';
+const jobName = 'ExpandJobData';
+let queue: Queue<ExpandJobData> | undefined = undefined;
+let worker: Worker<ExpandJobData> | undefined = undefined;
+
+/**
+ * Initializes the ValueSet expand worker.
+ * Sets up the BullMQ job queue.
+ * Sets up the BullMQ worker.
+ * @param config - The Medplum server config to use.
+ */
+export function initExpandWorker(config: MedplumServerConfig): void {
```

Comment:
Thinking about this more... What is the motivation for using a BullMQ job for this?

Do we expect it to fail and retry?